deepdoc/pipeline_v2.py

```
@@ -678,344 +630,62 @@ def _print_scan(self, scan: RepoScan) -> None:
-                # Rate limit between pages
-                if i < total - 1 and i % self.batch_size == self.batch_size - 1:
-                    time.sleep(RATE_LIMIT_PAUSE)
-
-        if failed > 0:
-            console.print(f"[yellow]⚠ {failed} page(s) failed[/yellow]")
-
-        return generated
-
-    def _generate_single_page(self, page, scan, plan) -> str:
-        """Generate a single doc page with retry logic.
+    def _apply_glossary_links(self, plan: DocPlan) -> None:
```

Ruff (0.16.1)

[error] Undefined name `DocPlan`

(F821)